src/MongoDB/Manager.c

```diff
@@ -200,6 +200,15 @@ static void php_phongo_manager_prep_uri_options(zval *options TSRMLS_DC) /* {{{
 				continue;
 			}
 
+            if (!strcasecmp(ZSTR_VAL(string_key), MONGOC_URI_READPREFERENCE)) {
+				ZVAL_DEREF(option);
+				SEPARATE_ZVAL_NOREF(option);
+				if (Z_TYPE_P(option) != IS_STRING) {
+                    phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Unsupported readPreference type, a string must be given");
```

Comment:
> I see, so basically the problem of the serialization was not with MongoDB\Driver\ReadPreference

It was technically due to ReadPreference not supporting PHP serialization (to be fixed by [PHPC-850](https://jira.mongodb.org/browse/PHPC-850)).

> so something like this 'readpreference' => new MongoDB\Driver\ReadPreference(MongoDB\Driver\ReadPreference::RP_SECONDARY), should not cause any serialization exceptions.

If PHPC-850 was implemented, we would get beyond the point where the connection URI options are serialized to a string for the hash string. A ReadPreference object inside of the options array would then get serialized to a BSON document according to its `bsonSerialize()` method, and I'd expect validation logic for PHPC-887 to throw an exception because a string value was expected for "readPreference".

> `php_phongo_apply_options_to_uri` or `php_phongo_apply_rp_options_to_uri`?

The read preference URI options are only handled in `php_phongo_apply_rp_options_to_uri()`. In https://github.com/mongodb/mongo-php-driver/pull/601#discussion_r123811088, I gave an example of how code inside `php_phongo_apply_options_to_uri()` should be changed. The read concern, read preference, and write concern options have their own functions, but all other URI options are handled by `php_phongo_apply_options_to_uri()`. You can start with any of these functions, but I think the general `php_phongo_apply_options_to_uri()` function may be best since you have an example already from my earlier comment.